devops/jobs/CreateSandbox.groovy

```diff
@@ -105,9 +105,9 @@ class CreateSandbox {
                          One reason you might want to override this field is if you are building a sandbox for review or a specific task. \
                          If setting this, you probably want to also set name_tag below. \
                          For example, if you are building a sandbox for pull request 1234 put in 'pr1234' which will setup the sandbox <i>pr1234.sandbox.edx.org</i>.<br /> \
-                         <b>Do not use hyphens or underscores here.</b> <br /> \
```

Comment:
Though I know we can use hyphens for some builds I am curious if the variable is sometimes passed into something that expects it to not have hyphens.  That said, I like hyphens so I support this change.

Reply:
Are we sure about this? I think it may get used for dns which means you can't use underscores. Do we know why this was documented in the first place and if the issue has been resolved?

Reply:
My Github username has a hyphen in it, which is how I know that hyphens aren't a problem here -- I can use the defaults and still get a working sandbox. Underscores are probably still a problem, because URLs can't have underscores in the host component, but I've left that warning in there.

Reply:
My sandboxes converge, but is it possible that something optional is breaking that I don't notice.

Looks like this was actually added relatively recently: https://github.com/edx/jenkins-job-dsl/pull/765 I'll ask in there to see if it's something that's still reproducible.

Reply:
I don't know how to reproduce the conditions Kyle mentions, so I'm reverting this part but explaining it a little.